src/librustc_codegen_llvm/mir/place.rs

```diff
@@ -174,7 +174,12 @@ impl PlaceRef<'ll, 'tcx> {
         let cx = bx.cx;
         let field = self.layout.field(cx, ix);
         let offset = self.layout.fields.offset(ix);
-        let align = self.align.min(self.layout.align).min(field.align);
+        let mut real_field_align = self.align.min(self.layout.align).min(field.align);
+        if offset.abi_align(real_field_align) > offset {
```

Comment:
Since this code is already duplicated twice in rustc and will get a third duplication in miri, I think this warrants a method on `Align` taking a field alignment and a field offset and producing the real alignment.

Reply:
That's not enough, sadly, unless you mean starting from `self.align.min(self.layout.align)` and then using the method *on that*. I think I like that, and I can fix the `FIXME` too.

Reply:
Yes that's what I meant.